### PR TITLE
AG-13115 Fallback to selection column if not grouping

### PR DESCRIPTION
**Pull request by eliasmalik**
See https://ag-grid.atlassian.net/browse/AG-13115

### CODE REVIEW

File: packages/ag-grid-community/src/selection/selectAllFeature.ts [hunk 1]
```diff
@@ -203,7 +203,10 @@ export class SelectAllFeature extends BeanStub {
             const isAutoCol = isColumnGroupAutoCol(column);
             // default to displaying header checkbox in the selection column
             const location = _getCheckboxLocation(rowSelection);
-            if ((location === 'autoGroupColumn' && isAutoCol) || (location === 'selectionColumn' && isSelectionCol)) {
+            if (
+                (location === 'autoGroupColumn' && isAutoCol) ||
+                (isSelectionCol && beans.selectionColSvc?.isSelectionColumnEnabled())
```
Comment (AndrewGlazier) [inline packages/ag-grid-community/src/selection/selectAllFeature.ts:208]: Is this selection col check needed? Wouldn't the selection col be hidden if location==='autoGroupColumn'?

File: packages/ag-grid-community/src/columns/selectionColService.ts [hunk 2]
```diff
@@ -87,14 +88,17 @@ export class SelectionColService extends BeanStub implements NamedBean {
...
+        const { gos, beans } = this;
         const rowSelection = gos.get('rowSelection');
         if (typeof rowSelection !== 'object') {
             return false;
         }
 
-        if (rowSelection.checkboxLocation === 'autoGroupColumn') {
+        if (
+            rowSelection.checkboxLocation === 'autoGroupColumn' &&
+            beans.visibleCols.allCols.some(isColumnGroupAutoCol)
```
Comment (AndrewGlazier) [inline packages/ag-grid-community/src/columns/selectionColService.ts]: Not certain, but autoColsSvc might be a better check here as it has _just_ the auto cols, so probably a faster look up

Comment (eliasmalik) [reply packages/ag-grid-community/src/columns/selectionColService.ts]: I think we can use `autoColsSvc` assuming there's no way to hide a group column?

File: packages/ag-grid-community/src/selection/selectAllFeature.ts (hunk 1 above)
Comment (eliasmalik) [reply packages/ag-grid-community/src/selection/selectAllFeature.ts:208]: We're using the selection col check in the case where `location != 'autoGroupColumn'`

File: packages/ag-grid-community/src/columns/selectionColService.ts (hunk 2 above)
Comment (AndrewGlazier) [reply packages/ag-grid-community/src/columns/selectionColService.ts]: We don't currently allow it so should be 👍

**Pull request merged**